### PR TITLE
Get some Prettier action going

### LINKED ISSUE #2142: Integrate Prettier into project
We need to get [Prettier](https://github.com/prettier/prettier) integrated into our project and integrated into our repo.

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
This PR fixes #2142

- [x] Add `Prettier` to project dependencies
- [x] Add `Prettier` to recommended extensions for VS Code
- [x] Add `Prettier` to VS Code workspace settings
- [x] ~Setup integrations on GitHub to automagically prettify all committed code~
- [x] Fail build in CI if things aren't pretty
- [x] Prettify entire project
- [x] Remove rules from TS Lint that will no longer be needed

## Important Info!!!

This PR adds some new scripts to prettify and to check the pretties of code.

If I have broken your branch run `npm run prettify`.
If you want to check what files are in need of some TLC, run `node script/is-it-pretty`

### CODE REVIEW

File: app/src/lib/dispatcher/actions.ts [hunk 1]
```diff
@@ -50,7 +50,13 @@ export interface IUpdateRepositoryPathAction {
   readonly path: string
 }
 
-export type Action = IGetAccountsAction | IGetRepositoriesAction |
-                     IAddRepositoriesAction | IUpdateGitHubRepositoryAction |
-                     IRemoveRepositoriesAction | IAddAccountAction | IRemoveAccountAction |
-                     IUpdateRepositoryMissingAction | IUpdateRepositoryPathAction
+export type Action =
+  | IGetAccountsAction
```
Comment (j-f1) [inline app/src/lib/dispatcher/actions.ts]: Are you sure this is valid syntax?

Comment (j-f1) [inline app/src/lib/dispatcher/actions.ts]: 😮 Turns out that’s [completely valid](http://www.typescriptlang.org/play/index.html#src=type%20foo%20%3D%20%7C%20string%20%7C%20number%20%7C%20boolean%0D%0Avar%20x%3A%20foo%20%3D%20true)

File: app/src/ui/preferences/git.tsx [hunk 2]
```diff
@@ -12,22 +12,14 @@ interface IGitProps {
...
   public render() {
     return (
       <DialogContent>
         <Row>
-          <TextBox label="Name" value={this.props.name} onChange={this.onNameChanged} autoFocus />
+          <TextBox label="Name" value={this.props.name} onChange={this.props.onNameChanged} autoFocus />
         </Row>
         <Row>
-          <TextBox label="Email" value={this.props.email} onChange={this.onEmailChanged} />
+          <TextBox label="Email" value={this.props.email} onChange={this.props.onEmailChanged} />
```
Comment (j-f1) [inline app/src/ui/preferences/git.tsx]: This should be `onValueChange`. Same above.

**Comment by joshaber** [comment]
Looks like tslint is suuuuuuuuper upset about some things 🙈

**Comment by iAmWillShepherd** [comment]
❗️ ❗️ ❗️ ❗️ ❗️

File: .vscode/settings.json [hunk 3]
```diff
@@ -15,5 +15,9 @@
         "**/out": true,
         "tslint-rules/**/*.js": true
     },
-    "editor.tabSize": 2
+    "editor.tabSize": 2,
+    "prettier.semi": false,
+    "prettier.singleQuote": true,
+    "prettier.trailingComma": "es5",
+    "prettier.printWidth": 100
```
Comment (joshaber) [inline .vscode/settings.json]: Should this be 80 to match what we give `prettier`?

File: script/is-it-pretty.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  const uglyFiles = [];
+  const matchCount = matches.length;
+
+  for (const match of matches) {
+    const fileContents = fs.readFileSync(match, 'utf8');
+    const isPretty = prettier.check(fileContents, prettierOptions);
+
+      if (!isPretty) {
```
Comment (joshaber) [inline script/is-it-pretty.js]: Indentation here is funny :trollface:

File: script/is-it-pretty.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        uglyFiles.push(match);
+      }
+  }
+
+  if (uglyFiles.length === 0) {
+    console.log("This is some pretty code");
+  } else {
+    console.log(`${uglyFiles.length} out of ${matchCount} code files are ugly. Please prettify the following files:`);
```
Comment (joshaber) [inline script/is-it-pretty.js]: Nice 🤘

File: script/is-it-pretty.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+'use strict'
+
+const prettier = require('prettier');
+const glob = require('glob');
+const fs = require('fs');
+const globPattern = 'app/{src, test}/**/*.{ts, tsx}'
+const prettierOptions = {
```
Comment (joshaber) [inline script/is-it-pretty.js]: I don't love how many places we have to duplicate our options. Can we just use the defaults?

File: script/is-it-pretty.js (hunk 5 above)
Comment (joshaber) [inline script/is-it-pretty.js]: Could we include the command they need to run? `npm run prettify` or whatever?

File: .vscode/settings.json (hunk 3 above)
Comment (joshaber) [inline .vscode/settings.json]: Oh maybe I misunderstood this... so this config determines how we see the files in our editor, and then when we commit it prettifies them?

Comment (iAmWillShepherd) [inline .vscode/settings.json]: Yes, but I'm down to make them all the same if it matters much.

File: script/is-it-pretty.js (hunk 6 above)
Comment (iAmWillShepherd) [inline script/is-it-pretty.js]: Unfortunately no, the defaults don't like typescript or other things our linters like.

File: script/is-it-pretty.js (hunk 5 above)
Comment (iAmWillShepherd) [inline script/is-it-pretty.js]: Sure.

File: .vscode/settings.json (hunk 3 above)
Comment (joshaber) [inline .vscode/settings.json]: No that makes sense 👍

File: script/is-it-pretty.js (hunk 6 above)
Comment (joshaber) [inline script/is-it-pretty.js]: 😭

**Comment by iAmWillShepherd** [comment]
💠

**Pull request merged**